sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/ExternalAndroidStorageAccessCheck.kt

```diff
@@ -0,0 +1,62 @@
+package org.sonarsource.kotlin.checks
+
+import org.jetbrains.kotlin.descriptors.PropertyDescriptor
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.psi.KtNameReferenceExpression
+import org.jetbrains.kotlin.psi.KtReferenceExpression
+import org.jetbrains.kotlin.psi2ir.unwrappedGetMethod
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.calls.model.ResolvedCall
+import org.sonar.check.Rule
+import org.sonarsource.kotlin.api.CallAbstractCheck
+import org.sonarsource.kotlin.api.FunMatcher
+import org.sonarsource.kotlin.plugin.KotlinFileContext
+
+private const val MESSAGE = "Accessing Android external storage is security-sensitive"
+
+private val HOTSPOT_FUNS = listOf(
+    FunMatcher(qualifier = "android.os.Environment") {
+        withNames(
+            "getExternalStorageDirectory",
+            "getExternalStoragePublicDirectory"
+        )
+    },
+    FunMatcher(qualifier = "android.content.Context") {
+        withNames(
+            "getExternalFilesDir",
+            "getExternalFilesDirs",
+            "getExternalCacheDir",
+            "getExternalCacheDirs",
+            "getExternalMediaDirs",
+            "getObbDir",
+            "getObbDirs",
+        )
+    }
+)
+
+private val HOTSPOT_PROPS = listOf(
+    "externalCacheDir",
+    "externalCacheDirs",
+    "externalMediaDirs",
+    "obbDir",
+    "obbDirs",
+)
+
+@Rule(key = "S5324")
+class ExternalAndroidStorageAccessCheck : CallAbstractCheck() {
+    override fun functionsToVisit() = HOTSPOT_FUNS
+
+    override fun visitFunctionCall(callExpression: KtCallExpression, resolvedCall: ResolvedCall<*>, kotlinFileContext: KotlinFileContext) {
+        val calleeExpression = callExpression.calleeExpression ?: return
```

Comment:
I think it's safe to use `callExpression.calleeExpression!!` when you've already matched the `FunMatcher`. I can't imagine the situation when it's null in this context.